packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -123,19 +181,78 @@ class DatePickerInput extends React.Component {
     }
   }
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
-      return;
+  handleOnKeyDown(event) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
     }
+  }
+
+  handleDayChange(event) {
+    this.handleDateChange(event, DateUtil.inputType.DAY);
+  }
+
+  handleMonthChange(event) {
+    this.handleDateChange(event, DateUtil.inputType.MONTH);
+  }
+
+  handleYearChange(event) {
+    this.handleDateChange(event, DateUtil.inputType.YEAR);
+  }
 
-    if (this.props.onChange) {
-      this.props.onChange(event);
+  handleOnFocus(event) {
+    this.setState({ isFocused: true });
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
     }
   }
 
-  handleOnKeyDown(event) {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
+  handleOnBlur(event) {
+    this.setState({ isFocused: false });
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleDateChange(event, type) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    if (type === DateUtil.inputType.DAY) {
+      this.setState({
+        day: inputValue,
+      });
+    } else if (type === DateUtil.inputType.MONTH) {
+      this.setState({
+        month: inputValue,
+      });
+    } else if (type === DateUtil.inputType.YEAR) {
+      this.setState({
```

Comment:
Is there any worry on update this state before the onChange value is called? The state update will cause a re-render